pet-store-react-hooks/src/PetModal/PetModal.test.tsx

```diff
@@ -0,0 +1,771 @@
+
+    const ageInput = await screen.findByLabelText('Age', {
+      selector: 'input',
+    });
+    expect(ageInput).toBeInTheDocument();
+    expect(ageInput).toBeEnabled();
+
+    // Question: Found a label with the text of: Has health problems, however the element associated with this label (<input />) is non-labellable
+    // Why do i get this error when i put 'checkbox' value for selector?
+    const hasHealthProblemsCheck = await screen.findByLabelText(
+      'Has health problems',
+      {
+        selector: 'input',
+      }
+    );
+    expect(hasHealthProblemsCheck).toBeInTheDocument();
+    expect(hasHealthProblemsCheck).toBeEnabled();
+
+    const notesInput = await screen.findByLabelText('Notes', {
+      selector: 'textarea',
+    });
+    expect(notesInput).toBeInTheDocument();
+    expect(notesInput).toBeEnabled();
+
+    const addedDateInput = await screen.findByLabelText('Added date', {
+      selector: 'input',
+    });
+    expect(addedDateInput).toBeInTheDocument();
+    expect(addedDateInput).toBeEnabled();
+    expect(addedDateInput).toHaveValue(toInputDate(new Date()));
+
+    const saveButton = await screen.findByRole('button', {
+      name: 'Save',
+    });
+    expect(saveButton).toBeInTheDocument();
+
+    const cancelButton = await screen.findByRole('button', {
+      name: 'Cancel',
+    });
+    expect(cancelButton).toBeInTheDocument();
+  });
+
+  test('When new pet is added, the modal state is changed to View and all inputs are locked', async () => {
+    const onClose = jest.fn();
+    const onModified = jest.fn();
+
+    const user = userEvent.setup();
+
+    render(
+      <PetModal
+        petId={undefined}
+        petKinds={petKinds}
+        petKindsMap={petKindsMap}
+        onClose={onClose}
+        onModified={onModified}
+      />
+    );
+
+    const nameInput = await screen.findByLabelText('Name', {
+      selector: 'input',
+    });
+    await user.type(nameInput, 'Test pet');
+
+    const kindSelectInput = await screen.findByLabelText('Kind', {
+      selector: 'select',
+    });
+    await user.selectOptions(kindSelectInput, '1');
+
+    const ageInput = await screen.findByLabelText('Age', {
+      selector: 'input',
+    });
+    await user.type(ageInput, '5');
+
+    const hasHealthProblemsCheck = await screen.findByLabelText(
+      'Has health problems',
+      {
+        selector: 'input',
+      }
+    );
+    await user.click(hasHealthProblemsCheck);
+
+    const notesInput = await screen.findByLabelText('Notes', {
+      selector: 'textarea',
+    });
+    await user.type(notesInput, 'Test notes');
+
+    // Question: Looked around 30 minutes but couldnt find a way to set the value with user events...
+    const addedDateInput = await screen.findByLabelText('Added date', {
+      selector: 'input',
+    });
+    const date = new Date();
+    date.setDate(date.getDate() + 1);
+    fireEvent.change(addedDateInput, {
```

Comment:
Why not just .type(...) ?

Reply:
Found similar issue about typing in date fields

> For anyone who happens upon this with the same issue I had: you may find that userEvent.type() may not fire expected events (and hence may not correctly update bound data) if a date input is already populated. This is because, like a text input, it tries to append to the existing value — which a date field doesn't allow. Need to userEvent.clear() it first

The value is different now after i clear the input and type in it
```
const addedDateInput = await screen.findByLabelText('Added date', {
  selector: 'input',
});
await user.clear(addedDateInput);
const date = new Date();
date.setDate(date.getDate() + 1);
await user.type(addedDateInput, toInputDate(date));
```